### PR TITLE
Fix test cleanup

### LINKED ISSUE #2602: [Bug] microVM cleanup not triggered in the CI tests using logger thread
# Describe the bug

By default, the instantiation of a microVM object will also create and start a thread that receives [as reference the microVM object](https://github.com/firecracker-microvm/firecracker/blob/main/tests/framework/microvm.py#L805). This results in the microVM destructor never being called and thus cleanup is skipped.

## To Reproduce

```
./tools/devtool shell -p && cd tests/`
# check the contents of the /srv/ directory (default location for test resources)
du -sh /srv/
pytest integration_tests/functional/test_api.py::test_negative_api_lifecycle
du -sh /srv/*
# 387M	/srv/jailer
# the jailer directory has not gone through the cleanup process
```

## Expected behaviour

The `/srv/jailer/firecracker` exists but does not contain any microvm leftover resources.

## Environment

No specific environment needed.

## Additional context

Move the logging thread at the microVM level or one level above it so that we do not need to reference it in the function running the logging thread. This would also help us with being able to check the microVM logs after the destruction of the microVM.

## Checks

- [x] Have you searched the Firecracker Issues database for similar problems?
- [x] Have you read the existing relevant Firecracker documentation?
- [x] Are you certain the bug being reported is a Firecracker issue?

### PR DESCRIPTION
**Pull request by alindima**
# Reason for This PR

Fixes #2602

## Description of Changes

See commit messages.

- [ ] This functionality can be added in [`rust-vmm`](https://github.com/rust-vmm).

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any API changes are reflected in `firecracker/swagger.yaml`.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

### CODE REVIEW

File: tests/integration_tests/performance/test_process_startup_time.py [hunk 1]
```diff
@@ -22,6 +24,19 @@ def test_startup_time_new_pid_ns(test_microvm_with_api):
...
+    assert os.path.exists(pid_file_path)
+
+    # Read the PID stored inside the file.
+    with open(pid_file_path) as file:
+        fc_pid = int(file.readline())
+    file.close()
+
+    # We need to explicitly kill the Firecracker pid, since it's different from
+    # the jailer pid which gets killed on object destruction.
+    utils.run_cmd(f'kill -9 {fc_pid}', ignore_return_code=True)
```
Comment (acatangiu) [inline tests/integration_tests/performance/test_process_startup_time.py:38]: Should we just add this in the standard microvm destructor?

Look for pidfile, if found `kill -9 $(cat $pidfile)`.

Would cover future tests as well.

Comment (alindima) [reply tests/integration_tests/performance/test_process_startup_time.py:38]: good idea. done

**Review by acatangiu — APPROVED** [review]
Looks good! Does this PR fix cleanup in tests using `MicrovmBuilder` too?

**Review by georgepisaltu — APPROVED** [review]
LGTM!

**Comment by alindima** [comment]
> Looks good! Does this PR fix cleanup in tests using `MicrovmBuilder` too?

Yes, by adding the `weakref` on the logger thread it breaks the dependency that caused the destructor to not run

**Pull request merged**